packages/swapper/src/swappers/zrx/zrxApproveInfinite/zrxApproveInfinite.ts

```diff
@@ -9,11 +10,21 @@ export async function zrxApproveInfinite<T extends EvmSupportedChainIds>(
   { quote, wallet }: ApproveInfiniteInput<T>
 ) {
   try {
+    const approvalQuote = {
```

Comment:
I suppose we can move forward with this, but fee management in swapper is def a bit of a mess... Another thought is we now have `approvalFee` on the `feeData.chainSpecific` data that we can garner the approval `gasLimit` from instead of writing over quote payloads.

May just want to wait until doing dynamic approval fee estimates though

Reply:
agreed this is an improvement but let's not tackle dynamic fees here